### ISSUE TITLE
PDF File does not retain its semantic meaning in the 'Term of use' page which was there before uploading that PDF File.

### ISSUE THREAD

**Issue by msftedad** [mozilla/pdf.js#16130]
[Responsible Computing User Agreement 5-4-2022 (1) (1) (1).pdf](https://github.com/mozilla/pdf.js/files/10919625/Responsible.Computing.User.Agreement.5-4-2022.1.1.1.pdf)

https://user-images.githubusercontent.com/93735775/223690594-eaebdb8e-aa5c-4b71-ac04-0ab33d202bf6.mp4

**Configuration:**
OS: Window 11 22H2 OS Build 25179.1000
Browser: Microsoft Edge Version 107.0.1402.2 (Official build) dev (64-bit)
URL: [https://mozilla.github.io/pdf.js/web/viewer.html](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fmozilla.github.io%2fpdf.js%2fweb%2fviewer.html&data=05%7c01%7cv-knarang%40microsoft.com%7c181f1d697db545eef23208db133bed47%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638124920433241849%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=wjI9a17lNu8UYT6ZF2xNzPdTJar1W56q3rVbB4gJSUA%3D&reserved=0)
Screen reader: Narrator

**Steps to reproduce:**
1. Open this PDF viewer [https://mozilla.github.io/pdf.js/web/viewer.html](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fmozilla.github.io%2fpdf.js%2fweb%2fviewer.html&data=05%7c01%7cv-knarang%40microsoft.com%7c181f1d697db545eef23208db133bed47%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638124920433241849%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=wjI9a17lNu8UYT6ZF2xNzPdTJar1W56q3rVbB4gJSUA%3D&reserved=0) and upload attached pdf with 2.Heading and links using 'Open File' option.
3.Navigate through links and headings and observe the issue.

**Actual Result:**
PDF File does not retain its semantic meaning in the 'Term of use' page which was there before uploading that PDF File. Some of the issues are: Screen Reader does not announce visible name of Links when user navigate using 'tab' key or 'k' key.
PDF file does not have any heading. Both issues do not repo on same PDF File before uploading.

**Expected Result:**
PDF File should retain its semantic meaning in the 'Term of use' page which was there before uploading that PDF File.

**Notes:**
From Dev: External issue.

Terms of Use uses an open source pdf converter to produce a universally viewable html page from the given pdf document.  Comparing the ToU experience with a browser's native implementation for pdf viewing does not match the restrictions our product has to be able to be viewed in embedded sign-in flows.

Can guidance be updated for these accessibility tests to ensure tests are run against  https://mozilla.github.io/pdf.js/web/viewer.html .  This is the same viewer library we use.  If the bug is present here, then the bug needs to be considered external 3rd party.

**Comment by v-rpundir** [comment]
Hi @timvandermeij Any fix ETA for this bug as it came from one of Customer using PDFJS?

Thanks :):

**Comment by msft-rahul** [comment]
@timvandermeij Could you please provide fix ETA for this bug as it is impacting our product end-user/customers who is using this PDF viewer.

Thanks :):

**Comment by CarlyGerard** [comment]
I can confirm this bug using a different PDF [(degree confirmation PDF example)](https://github.com/mozilla/pdf.js/files/11534623/test-pdf-degree-confirmation.pdf). If you open the document in Adobe and view the tag structure you should see the page title marked as an H1:
![Adobe's tag structure reveals '2020-2021 Bachelor’s Degree Confirmation' marked as H1](https://github.com/mozilla/pdf.js/assets/27382956/9ea00dba-b188-460a-a600-818b8ef8cf9d)
When opening and viewing with PDF.js at [https://mozilla.github.io/pdf.js/web/viewer.html](https://mozilla.github.io/pdf.js/web/viewer.html), that same text is marked up as as `<span>`, losing its semantics:
![The degree confirmation PDF is loaded in the repo's viewer page](https://github.com/mozilla/pdf.js/assets/27382956/1473ba6b-cb1f-466e-9feb-a4a2f5d6a97b)
!['2020-2021 Bachelor’s Degree Confirmation' text now wrapped in nested span tags](https://github.com/mozilla/pdf.js/assets/27382956/d516c9c1-023c-4a3b-811b-92a35f1777aa)

**Comment by rahulpundir011** [comment]
Yes @CarlyGerard that is the issue. PDF losses its semantics when open inside https://mozilla.github.io/pdf.js/web/viewer.html

**Comment by calixteman** [comment]
@CarlyGerard the pdf you linked is no more here.
@msftedad do you have the same issue in using Firefox pdf viewer ?

I just tested myself in Firefox nightly with NVDA and I'm able to go from a heading to another in using the H key.
In Firefox, it's possible to see accessibility properties in the devtools (right click on a the pdf and then open `Inspect Accessibility Properties`) and the pdf in the first comment I can clearly see that the semantics seem to there.

@CarlyGerard, there are some a11y info in the `structTree` element under the canvas element in each page. This `structTree` have some elements and these ones have some aria-owns.

**Comment by rahulpundir011** [comment]
Hi @calixteman Heading issue does not repro on Firefox or pdf viewer(https://mozilla.github.io/pdf.js/web/viewer.html) now but second issue which I mentioned in the bug that when user navigate through links using 'k' or 'tab' key. Screen Reader does not announce link name could your team also fix that.

Links issue does not repro on Edge pdf viewer.

CC- @msftedad 

Thanks :):

**Comment by rahulpundir011** [comment]
@calixteman Any Update for link Issue?

**Comment by msft-rahul** [comment]
Hi @calixteman I am from Dependent Product team, heading issue does not repro on Firefox or pdf viewer(https://mozilla.github.io/pdf.js/web/viewer.html) now but second issue which I mentioned in the bug that when user navigate through links using 'k' or 'tab' key. Screen Reader does not announce link name could your team also fix that.

Links issue does not repro on Edge pdf viewer.

CC- @msftedad

Thanks :):

**Comment by msft-rahul** [comment]
Hi @calixteman I am from Dependent Product team, Heading issue does not repro on Firefox or pdf viewer(https://mozilla.github.io/pdf.js/web/viewer.html) now but second issue which I mentioned in the bug that when user navigate through links using 'k' or 'tab' key. Screen Reader does not announce link name could your team also fix that.

Links issue does not repro on Edge pdf viewer.

CC- @msftedad

Thanks :):

**Comment by msftedad** [comment]
Hi @calixteman, Could you please let us know tentative fix ETA for this issue.

**Comment by msftedad** [comment]
Hi, Any update on this issue? Please provide the ETA to fix this issue.

**Comment by msftedad** [comment]
Hi @calixteman, @CarlyGerard, Hi, Any update on this issue? Please provide the ETA to fix this issue.

**Comment by timvandermeij** [comment]
@msftedad Please keep in mind that this is an open source project, frequented by many volunteers, rather than paid support where one could request faster resolution of an issue. Therefore, please refrain from pinging people and asking for ETAs. The issue is logged and if there are updates they will be posted on this issue.

Anyone is free to work on this issue, so if you or someone else can contribute a patch to speed up the resolution of the issue we'll be happy to review it.

**Comment by Snuffleupagus** [comment]
Given https://github.com/mozilla/pdf.js/issues/16130#issuecomment-1559906521 this might even be fixed already!?

Furthermore, considering that: Microsoft Edge isn't really a development target for this project, this issue contains some amount of unrelated content, and finally the persistent spamming it's not entirely clear that it's helpful/meaningful to keep this open.

**Comment by msftedad** [comment]
Hi @Snuffleupagus, We are checking on this, we will update you soon.
